src/main/java/services/CategoryTimingHelper.java

```diff
@@ -0,0 +1,70 @@
+package services;
+
+import com.timgroup.statsd.StatsDClient;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import util.Constants;
+import util.FormplayerHttpRequest;
+import util.FormplayerRaven;
+import util.SimpleTimer;
+
+@Component
+public class CategoryTimingHelper {
+    @Autowired
+    private FormplayerHttpRequest request;
+
+    @Autowired
+    private StatsDClient datadogStatsDClient;
+
+    @Autowired
+    private FormplayerRaven raven;
+
+    public class RecordingTimer extends SimpleTimer {
```

Comment:
is there an advantage to putting this class definition within the `CategoryTimingHelper` as opposed to having it be in its own file?

Reply:
Yeah I don't know; I guess I was considering it more of an extension of `CategoryTimingHelper` than its own thing; for example when it takes a `CategoryTimingHelper parent` arg, it's doing so to gain access to `CategoryTimingHelper`'s private functions and (indirectly) its `Autowired` globals. You'd never have one without the other so this nested class seemed like the most obvious way to show that.

`SimpleTimer` on the other hand is useful and used elsewhere, so that is its own separate class.

I'm of course carrying a lot of my sensibilities about these things over form python where it's customary to put more things in a single file, but I guess I figured if inner classes are ever acceptable in Java, this would be one such case.